### DISCUSSION TITLE
Is there a way to query the robot arm/s motion status (moving or not) using this driver

### DISCUSSION THREAD

**Discussion by bhomaidan1990** [ros-industrial/abb_robot_driver#44] [Q&A] [answered]
Hi,
I do want to query the status of each arm, or at least one of them at once as fast as possible using this driver,
I was using the `RWS` HTTP server before `http://host/rw/rapid/execution` which has some drawbacks.

Can you please tell me how can I query the robot motion status (arm moving/not moving) using this driver? thanks in advance.

**Comment by gavanderhoorn** [comment, marked as answer]
IIRC, there isn't anything specifically publishing something which directly tells you whether a mechanical unit is "moving", but the `egm_convergence_met` field (in [EGMChannelState](https://github.com/ros-industrial/abb_robot_driver_interfaces/blob/5376912f7c676754033e5ff9c652fb94307c371a/abb_egm_msgs/msg/EGMChannelState.msg#L37)) comes close.

That field should be `false` when a new offset has been applied using EGM -- which in the case of `abb_robot_driver` would be the case when it commands a mechanical unit to move.

Note that when that field transitions to `true` depends on the convergence settings, but that's mostly a problem/challenge when trying to determine a mechanical unit has stopped moving.

**Reply by bhomaidan1990** [reply]
@gavanderhoorn thanks for your kind reply, I'm moving the robot using **rapid routines** through calling them via **service calls**, rather than activating EGM channels, will this approach be feasible in my case?

**Reply by gavanderhoorn** [reply]
No.

I must admit that when you post something here, on the `abb_robot_driver` repository issue tracker / discussion forum, I assume you're actually *using* `abb_robot_driver`.

If you are not, then your question is off-topic.

**Reply by bhomaidan1990** [reply]
"*I assume you're actually using abb_robot_driver*", I'm using the driver service calls to interact with rapid routines (passing variables, and running the routines) but still I'm not using the `EGM` now to control the motion of the robot (at least in the current phase).

**Reply by gavanderhoorn** [reply]
As discussed in #40, "using `abb_robot_driver`" to *me* would imply you're actually controlling motion with it. As that's what one would typically understand "using a ROS robot driver" to mean.

It's OK if you only use a part of the driver, but that would be something to clarify, as it's an a-typical use.
